### DISCUSSION TITLE
Custom FileNamer, method originalFileName not working.

### DISCUSSION THREAD

**Discussion by EgoRoman** [spatie/laravel-medialibrary#2619] [Q&A] [answered]
I'm trying to create a custom class for naming files when uploading.
Using this docs https://spatie.be/docs/laravel-medialibrary/v9/advanced-usage/naming-files

Created class:

```
namespace App\Http;

use Spatie\MediaLibrary\MediaCollections\Models\Media;
use Spatie\MediaLibrary\Support\FileNamer\FileNamer;
use Spatie\MediaLibrary\Conversions\Conversion;

class ProductGalleryFileNamer extends FileNamer
{
  public function originalFileName(string $fileName): string
  {
    $mediaItem = Media::where('name', $fileName)->get();

    if ($mediaItem->count()) {
      $fileName = $fileName . '_' . $mediaItem->count(); // if file with same name exists then add suffix
    }

    return pathinfo($fileName, PATHINFO_FILENAME);
  }

  public function conversionFileName(string $fileName, Conversion $conversion): string
  {
    $strippedFileName = pathinfo($fileName, PATHINFO_FILENAME);

    $mediaItem = Media::where('name', $fileName)->get();

    if ($mediaItem->count()) {
      return "{$mediaItem->count()}-{$strippedFileName}-{$conversion->getName()}";
    } else {
      return "{$strippedFileName}-{$conversion->getName()}";
    }
  }

  public function responsiveFileName(string $fileName): string
  {
    return pathinfo($fileName, PATHINFO_FILENAME);
  }
}
```

Method 'conversionFileName' works perfect. 

I need a suffix to be added to the original name if a file with that name already exists, 
but method 'originalFileName' the method does not produce any effect.

**Comment by PaolaRuby** [comment, marked as answer]
that code is terrible, `Media::where('name', $fileName)->get()`, better `Media::where('name', $fileName)->count()`
you misunderstood how it works `originalFileName`
also you must use markdown correctly, **example**:
```php
public function responsiveFileName(string $fileName): string
{
    return pathinfo($fileName, PATHINFO_FILENAME);
}
```
Here is your mistake
https://github.com/spatie/laravel-medialibrary/blob/13514710bda92cdb0c3ead3728965d6773c9722b/src/Support/FileNamer/FileNamer.php#L10-L16

**Reply by EgoRoman** [reply]
No. That's not the point.
It seems to me that the method 'originalFileName' does not work at all.

For example.
This method works out and returns in the browser response the filename:
```
public function conversionFileName(string $fileName, Conversion $conversion): string
  {
    $strippedFileName = pathinfo($fileName, PATHINFO_FILENAME);

    echo $strippedFileName;
    die();
  }
```

But there is no reaction at all, the file is saved as usual, there is no stopping, and there is no output:
```
public function originalFileName(string $fileName): string
  {
    $extLength = strlen(pathinfo($fileName, PATHINFO_EXTENSION));
    $baseName = substr($fileName, 0, strlen($fileName) - ($extLength ? $extLength + 1 : 0));

    echo $baseName;
    die();
  }
```

It doesn't work that way either (https://github.com/spatie/laravel-medialibrary/blob/c0d9bcdfdd954523f817295d2fde73401e1a8e91/tests/TestSupport/RenameOriginalFileNamer.php):
```
public function originalFileName(string $fileName): string
    {
        return 'renamed_original_file';
    }
```

**Reply by PaolaRuby** [reply]
Tested and working as expected
https://github.com/spatie/laravel-medialibrary/blob/e0446d457772de8e9153e00f88806a2babca3bb3/tests/Feature/FileAdder/IntegrationTest.php#L446-L456
```batch
./vendor/bin/phpunit --filter=the_file_name_can_be_modified_using_a_file_namer
PHPUnit 9.5.10 by Sebastian Bergmann and contributors.

Runtime:       PHP 8.0.5
Configuration: /github/media/phpunit.xml.dist
.                                                                   1 / 1 (100%)
Time: 00:00.203, Memory: 24.00 MB
OK ( 1 test, 2 assertions)
```
I can see your mistake, can you see it?
>Here is your mistake

**Reply by EgoRoman** [reply]
I can't. =(

And why is there only such content in my vendor\spatie\laravel-medialibrary\src\Support\FileNamer\FileNamer.php?

```
<?php

namespace Spatie\MediaLibrary\Support\FileNamer;

use Spatie\MediaLibrary\Conversions\Conversion;
use Spatie\MediaLibrary\MediaCollections\Models\Media;

abstract class FileNamer
{
    abstract public function conversionFileName(string $fileName, Conversion $conversion): string;

    abstract public function responsiveFileName(string $fileName): string;

    public function temporaryFileName(Media $media, string $extension): string
    {
        return "{$this->responsiveFileName($media->file_name)}.{$extension}";
    }

    public function extensionFromBaseImage(string $baseImage): string
    {
        return pathinfo($baseImage, PATHINFO_EXTENSION);
    }
}

```

**Reply by PaolaRuby** [reply]
>And why is there only such content in my vendor\spatie\laravel-medialibrary\src\Support\FileNamer\FileNamer.php?

Ok that is weird, maybe you accidentally deleted that, just get a clean copy with composer `composer update`

**Reply by EgoRoman** [reply]
Nope, is the same after the update.
This is composer.lock:
```
"name": "spatie/laravel-medialibrary",
            "version": "9.0.0",
            "source": {
                "type": "git",
                "url": "https://github.com/spatie/laravel-medialibrary.git",
                "reference": "0ddacce06fbd33de321f01c1ad8c402cda9c6b10"
            },
```

**Reply by PaolaRuby** [reply]
we are on version [9.8.1](https://github.com/spatie/laravel-medialibrary/releases/tag/9.8.1), even your composer is wrongly configured

**Reply by EgoRoman** [reply]
Hallelujah. Installed the version 9,8,1 and now it works. Thanks.

But I used to install the command from the documentation (https://spatie.be/docs/laravel-medialibrary/v9/installation-setup):
```
composer require "spatie/laravel-medialibrary:^9.0.0"
```
And the version 9,0,0 was installed.

**Reply by PaolaRuby** [reply]
Must be `composer require spatie/laravel-medialibrary:"^9.0.0"`, documentation error
